In [ ]:
#dotnet tool install --global PowerShell                     #This is needed or the Automation namespace doesn't load in a Notebook
#Install-PSResource PowerShellForGitHub -Scope AllUsers -Verbose
#$Cred = New-Object System.Management.Automation.PSCredential "<username>",$(ConvertTo-SecureString '<PAT>' -asplaintext -force)
#Set-GitHubAuthentication -Credential $Cred 

In [1]:
$RepositoryOwner, $RepositoryName = 'PowerShell', 'PowerShell'#Change this to the target account and repo name. 
$ErrorActionPreference ='SilentlyContinue'                    #PowerShellForGitHub's update check function was causing an error with .Net interactive
$issues = Get-GitHubIssue -OwnerName $RepositoryOwner -RepositoryName $RepositoryName -State Open
if (-not $issues.Count) { Write-Host "No open issues found."; return }

$labelCounts = @{}
foreach ($issue in $issues) {
    foreach ($label in $issue.labels) {$labelCounts[$label.name]++}
}
$totalIssues = $issues.Count
$labelStats = $labelCounts.Keys | ForEach-Object {
    [PSCustomObject]@{
        Label      = $_
        Count      = $labelCounts[$_]
        Percentage = [math]::Round(($labelCounts[$_] / $totalIssues) * 100, 2)
    }
}

($labelStats).Where({$_.Percentage -gt 3})|Sort Percentage -Descending|FT #This is the table with count and percentage">3%" of which labels are applied to how many issues.


Label                  Count Percentage
-----                  ----- ----------
Needs-Triage             458     53.070
Issue-Enhancement        294     34.070
WG-Engine                137     15.870
Issue-Bug                 94     10.890
Review - Needed           87     10.080
Up-for-Grabs              77      8.920
WG-Cmdlets-Utility        69      8.000
WG-Cmdlets                62      7.180
WG-Language               48      5.560
In-PR                     46      5.330
WG-Cmdlets-Management     45      5.210
Issue-Question            43      4.980
Issue-Discussion          41      4.750
Extra Small               40      4.630
WG-Cmdlets-Core           35      4.060
WG-NeedsReview            34      3.940
WG-Interactive-Console    31      3.590



Error: Command failed: SubmitCode: $Cred = New-Object System.Management.Automation.PS ...

In [4]:
#Install-Module FormatMarkdownTable
$labelStats = $labelCounts.Keys | ForEach-Object {
    $labelUrl = "https://github.com/$RepositoryOwner/$RepositoryName/labels/$($_ -replace ' ', '%20')"
    [PSCustomObject]@{
        Label      = "[$_]($labelUrl)" 
        Count      = $labelCounts[$_]
        Percentage = [math]::Round(($labelCounts[$_] / $totalIssues) * 100, 2)
    }
}

($labelStats).Where({ $PSItem.Percentage -gt 3})|Sort Percentage -Descending|Format-MarkdownTableTableStyle


Count Label                                                                                            Percentage
----- -----                                                                                            ----------
  458 [Needs-Triage](https://github.com/PowerShell/PowerShell/labels/Needs-Triage)                         53.070
  294 [Issue-Enhancement](https://github.com/PowerShell/PowerShell/labels/Issue-Enhancement)               34.070
  137 [WG-Engine](https://github.com/PowerShell/PowerShell/labels/WG-Engine)                               15.870
   94 [Issue-Bug](https://github.com/PowerShell/PowerShell/labels/Issue-Bug)                               10.890
   87 [Review - Needed](https://github.com/PowerShell/PowerShell/labels/Review%20-%20Needed)               10.080
   77 [Up-for-Grabs](https://github.com/PowerShell/PowerShell/labels/Up-for-Grabs)                          8.920
   69 [WG-Cmdlets-Utility](https://github.com/PowerShell/PowerShell/labels/WG-Cmdlets-U

|Count|Label|Percentage|
|:--|:--|:--|
|458|[Needs-Triage](https://github.com/PowerShell/PowerShell/labels/Needs-Triage)|53.07|
|294|[Issue-Enhancement](https://github.com/PowerShell/PowerShell/labels/Issue-Enhancement)|34.07|
|137|[WG-Engine](https://github.com/PowerShell/PowerShell/labels/WG-Engine)|15.87|
|94|[Issue-Bug](https://github.com/PowerShell/PowerShell/labels/Issue-Bug)|10.89|
|87|[Review - Needed](https://github.com/PowerShell/PowerShell/labels/Review%20-%20Needed)|10.08|
|77|[Up-for-Grabs](https://github.com/PowerShell/PowerShell/labels/Up-for-Grabs)|8.92|
|69|[WG-Cmdlets-Utility](https://github.com/PowerShell/PowerShell/labels/WG-Cmdlets-Utility)|8|
|62|[WG-Cmdlets](https://github.com/PowerShell/PowerShell/labels/WG-Cmdlets)|7.18|
|48|[WG-Language](https://github.com/PowerShell/PowerShell/labels/WG-Language)|5.56|
|46|[In-PR](https://github.com/PowerShell/PowerShell/labels/In-PR)|5.33|
|45|[WG-Cmdlets-Management](https://github.com/PowerShell/PowerShell/labels/WG-Cmdlets-Management)|5.21|
|43|[Issue-Question](https://github.com/PowerShell/PowerShell/labels/Issue-Question)|4.98|
|41|[Issue-Discussion](https://github.com/PowerShell/PowerShell/labels/Issue-Discussion)|4.75|
|40|[Extra Small](https://github.com/PowerShell/PowerShell/labels/Extra%20Small)|4.63|
|35|[WG-Cmdlets-Core](https://github.com/PowerShell/PowerShell/labels/WG-Cmdlets-Core)|4.06|
|34|[WG-NeedsReview](https://github.com/PowerShell/PowerShell/labels/WG-NeedsReview)|3.94|
|31|[WG-Interactive-Console](https://github.com/PowerShell/PowerShell/labels/WG-Interactive-Console)|3.59|
